src/runtime_src/core/pcie/driver/linux/xocl/mgmtpf/mgmt-core.c

```diff
@@ -822,7 +841,12 @@ void xclmgmt_mailbox_srv(void *arg, void *data, size_t len,
 			mgmt_err(lro, "peer request dropped, wrong size\n");
 			break;
 		}
-		ret = xocl_icap_ocl_update_clock_freq_topology(lro, clk);
+
+		ret = xocl_has_icap(lro) ?
```

Comment:
It is better to check the return value (-ENODEV) instead of defining xocl_has_icap().
Why do we still keep xocl_icap_ocl_update_clock_freq_topology in icap?  Should it be better to call xocl_clock_update _freq in both cases?

Reply:
Adjust the code to check -ENODEV.
The xocl_icap_ocl_update_clock_freq_topology is still need here due to it checks xclbin min/max freq. After we can split the xclbin and icap, we can move this checking to better place.